### DISCUSSION TITLE
Database Design and how one learns enough to contribute to Arcade

### DISCUSSION THREAD

**Discussion by pedram80** [ArcadeData/arcadedb#648] [General]
Hi All,  I would love to contribute to ArcadeDB at some point but how does someone who is new to database design can get started ? There is a lot of resources out there but I love to see if there is some guidance I can follow that can help me speed through the process. My end goal would be to map core concepts to ArcadeDB implementation and be able to contribute . I have Java programming background with mostly backend implementation for web services but lack core knowlege on database design and distributed computing . Many resources out there are using low level languages like C or C++ and I could not find something with Java.  Any suggestion as too where I can learn the core concepts that are applied to ArcadeDB and start getting familiar with those implementations.

**Comment by lvca** [comment]
Hi @pedram80, we'd love to see your contributions. You can start from little things, like typos or small fixes, to move to something more complex and maybe one day you can get an issue on your own. There is no guided tour in the code base, some useful points:
- distributed: ArcadeDB uses Raft protocol, it's well documented on the internet if you want to see how it works
- indexes: LSM-Tree indexes are well documented on the internet, if you are interested in our optimized implementation the best way is to look into the code
- server: we use Undertow for HTTP and websocket. Undertow is quite well documented if you want to touch that part

**Reply by pedram80** [reply]
Thanks for the feedback . I have come across these resources that might be useful , I think LSM-Tree and Raft protocol much more complex topics 

https://ocw.mit.edu/courses/6-830-database-systems-fall-2010/pages/lecture-notes/
https://www.awelm.com/posts/simple-db/
https://github.com/LouisYLWang/SimpleDB
